plugins/rbac/src/hooks/useRoles.ts

```diff
@@ -70,42 +75,76 @@ export const useRoles = (
     permission: policyEntityUpdatePermission,
     resourceRef: policyEntityUpdatePermission.resourceType,
   });
+
+  useEffect(() => {
+    const fetchAllPermissionPolicies = async () => {
+      if (Array.isArray(roles)) {
+        try {
+          const updatedRoles: RoleWithConditionalPolicies[] = await Promise.all(
+            roles.map(async role => {
+              const conditionalPolicies = await rbacApi.getRoleConditions(
+                role.name,
+              );
+
+              return {
+                ...role,
+                conditionalPolicies: Array.isArray(conditionalPolicies)
+                  ? conditionalPolicies.length
+                  : 0,
+              };
+            }),
```

Comment:
@ciiay I will suggest that we dont do these computations inside `Promise.all` to reduce code complexity and increase readability. You can create an array of promises like this and then use `Promise.all`:
```
conditionsPromises = roles.map(role =>  rbacApi.getRoleConditions(role.name));
const conditions = await Promise.all(conditionsPromises);
```
After this you can have another function to map conditionsCount to respective roles using `roleEntityRef` param from the condition object. This will make the code a bit more linear as it would be divided into isolated steps.